packages/custom_lint_core/test/configs_test.dart

```diff
@@ -19,10 +20,61 @@ File createAnalysisOptions(String content) {
   return PhysicalResourceProvider.INSTANCE.getFile(ioFile.path);
 }
 
+Future<String> createTempProject(String projectName, String tempDirPath) async {
+  final projectPath = join(tempDirPath, projectName);
+
+  final dir = io.Directory(projectPath);
+  await dir.create();
+
+  final libPath = join(dir.path, 'lib');
+  await io.Directory(libPath).create();
+
+  final analysisOptionsPath = join(libPath, 'analysis_options.yaml');
+  await io.File(analysisOptionsPath).writeAsString('''
+custom_lint:
+  rules:
+    - from_package
+  ''');
+
+  return dir.path;
+}
+
+Future<void> patchPackageConfig(
+  String packageName,
+  String packagePath,
+) async {
+  final currentPackageConfig = await findPackageConfig(io.Directory.current);
+  if (currentPackageConfig == null) {
+    throw Exception('Could not find package config');
+  }
+
+  final patchedPackages = currentPackageConfig.packages.toList()
+    ..add(
+      Package(
+        packageName,
+        Uri.file('$packagePath/'),
+        packageUriRoot: Uri.parse('lib/'),
+      ),
+    );
+
+  final patchedPackageConfig = PackageConfig(
+    patchedPackages,
+    extraData: currentPackageConfig.extraData,
+  );
+
+  await savePackageConfig(
+    patchedPackageConfig,
+    io.Directory.current,
+  );
+
+  addTearDown(
+    () async => savePackageConfig(currentPackageConfig, io.Directory.current),
+  );
```

Comment:
Why are you wrapping it with `addTearDown`?

Reply:
because we should make this operation at the end of the test